File: Coding Interview Prep/abc_blocks.js
function canMakeWord(word) {
    word = word.toUpperCase();
    const blockArr =[["B", "O"], ["X", "K"], ["D", "Q"], ["C", "P"], ["N", "A"], 
               ["G", "T"], ["R", "E"], ["T", "G"], ["Q", "D"], ["F", "S"], ["J", "W"],
                ["H", "U"], ["V", "I"], ["A", "N"], ["O", "B"], ["E", "R"], ["S", "F"],
                ["L", "Y"], ["P", "C"], ["Z", "M"]];
    const newArr =[];
    for (let i = 0; i < word.length; i++){
        for (let arr of blockArr) {
             if (arr.includes(word.slice(i, i+1)) && !(newArr.includes(arr))) {
                newArr.push(arr);
                // break ensures loop stops once it adds a new value to the array
                break
             } 
        }
    // since each block represents a letter, the array of blocks needs to be the same length;
    // too short and it required duplicates that could not be provided (it can't be longer).
    } return newArr.length == word.length ? true : false
}

console.log(canMakeWord("snootz"));

File: Coding Interview Prep/abc_blocks.py
def can_make_word(word):
    word = word.upper()
    block_list = [["B", "O"], ["X", "K"], ["D", "Q"], ["C", "P"], ["N", "A"],
                  ["G", "T"], ["R", "E"], ["T", "G"], ["Q", "D"], ["F", "S"], ["J", "W"],
                  ["H", "U"], ["V", "I"], ["A", "N"], ["O", "B"], ["E", "R"], ["S", "F"],
                  ["L", "Y"], ["P", "C"], ["Z", "M"]]
    new_block_list = []
    i = 0
    while i < len(word):
        for ch in word:
            for block in block_list:
                if ch in block and not (block in new_block_list):
                    new_block_list.append(block)
                    i += 1
                    break
                else:
                    i += 1
    return True if len(new_block_list) == len(word) else False


print(can_make_word("snootz"))
